### PR TITLE
practices: Fix leaks in struct GraphicsPipelineCIs

### PR DESCRIPTION
**Pull request by jeremyg-lunarg**
Use optional rather than raw new to avoid having to remember to clean up this structure.

Fix leaks in test cases too.

### CODE REVIEW

File: tests/vklayertests_best_practices.cpp [hunk 1]
```diff
@@ -126,13 +126,13 @@ TEST_F(VkBestPracticesLayerTest, UseDeprecatedInstanceExtensions) {
     // Create a 1.0 vulkan instance and request an extension promoted to core in 1.1
     m_errorMonitor->ExpectSuccess(VK_DEBUG_REPORT_PERFORMANCE_WARNING_BIT_EXT);
     m_errorMonitor->SetUnexpectedError("UNASSIGNED-khronos-Validation-debug-build-warning-message");
-    VkApplicationInfo* new_info = new VkApplicationInfo;
+    std::unique_ptr<VkApplicationInfo> new_info(new VkApplicationInfo);
```
Comment (ncesario-lunarg) [inline tests/vklayertests_best_practices.cpp:129]: `layer_data::make_unique`?

**Review by ncesario-lunarg — APPROVED** [review]
:tada:

File: tests/vklayertests_best_practices.cpp (hunk 1 above)
Comment (jeremyg-lunarg) [reply tests/vklayertests_best_practices.cpp:129]: On second thought, I don't think it needs to be any kind of pointer.

Comment (ncesario-lunarg) [reply tests/vklayertests_best_practices.cpp]: Even better!

**Pull request merged**